features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/data/api.js

```diff
@@ -505,8 +505,55 @@ class API extends Resource {
     }
 
     /**
-     * Mock api call with delay
-     * todo: Replace with the rest api when available
+     * Get list of emails configured at Bot Detection --> Configure Emails
+     */
+    botDetectionNotifyingEmailsGet() {
+        return this.client.then((client) => {
+            return client.apis['default'].get_botData_getEmailList(
+                this._requestMetaData(),
+            );
+        });
+    }
+
+    addBotDetectionNotifyingEmail(email, callback = null) {
```

Comment:
Please don't use callback functions? no need to use callbacks with promises, We have used this pattern back then but realize there is no use of passing call back function as an argument to a function which returns a `Promise` object because `addBotDetectionNotifyingEmail` function callee have complete freedom to append any number of success (then) and failure (catch) handlers 😃

Reply:
Fixed all occurrences with commit: https://github.com/wso2/carbon-apimgt/pull/8591/commits/5138aa2a7e1b1611004d5260e67730dc11da6bb8